sys/debug/kgmon.py

```diff
@@ -0,0 +1,52 @@
+import gdb
+import sys
+
+
+from ctypes import *
+from struct import unpack
+from .cmd import SimpleCommand
+from .struct import GdbStructMeta
+
+
+class GmonOut():
+
+    def write_to_file(self, path="gmon.out"):
+        infer = gdb.inferiors()[0]
+        with open(path, "wb") as of:
+            self.write_header(infer, of)
+            self.write_tick_buffer(infer, of)
+            self.write_arc_info(infer, of)
+
+    def write_header(self, inferior, file):
+        gmonhdr_size = int(gdb.parse_and_eval('sizeof(_gmonhdr)'))
+        gmonhdr_p = gdb.parse_and_eval('&_gmonhdr')
+        memory = inferior.read_memory(gmonhdr_p, gmonhdr_size)
+        array = unpack('IIiiiiii', memory)
+        for i in array:
+            file.write(c_int(i))
+
+    def write_tick_buffer(self, inferior, file):
+        kcountsize = int(gdb.parse_and_eval('_gmonparam.kcountsize'))
+        kcount = gdb.parse_and_eval('_gmonparam.kcount')
+        memory = inferior.read_memory(kcount, kcountsize)
+        array = unpack(kcountsize * 'c', memory)
+        for i in array:
+            file.write(c_char(i))
```

Comment:
Might as well do it like this at this point:
```suggestion
        file.write(inferior.read_memory(kcount, kcountsize))
```